respect-lib-shared/src/commonMain/kotlin/world/respect/shared/domain/account/validateqrbadge/ValidateQrCodeUseCase.kt

```diff
@@ -0,0 +1,89 @@
+package world.respect.shared.domain.account.validateqrbadge
+
+import org.jetbrains.compose.resources.StringResource
+import world.respect.shared.generated.resources.Res
+import world.respect.shared.generated.resources.*
+
+data class QrValidationResult(
+    val isValid: Boolean = true,
+    val errorMessage: StringResource? = null
+) {
+    companion object {
+        val Valid = QrValidationResult(isValid = true)
+        val InvalidFormat = QrValidationResult(
+            isValid = false,
+            errorMessage = Res.string.qr_code_invalid_format
+        )
+        val InvalidSchoolUrl = QrValidationResult(
+            isValid = false,
+            errorMessage = Res.string.qr_code_must_start_with_school_url
+        )
+        val MissingBadgeSection = QrValidationResult(
+            isValid = false,
+            errorMessage = Res.string.qr_code_must_contain_respect_qr_badge
+        )
+    }
+}
+
+class ValidateQrCodeUseCase {
+
+    operator fun invoke(
+        qrCodeUrl: String,
+        schoolUrl: String?,
```

Comment:
The schoolUrl should not be a parameter to the invoke function. UseCase parameters that are the same should be constructor parameters, not parameters to the invoke function.

ValidateQrCodeUseCase should be scoped to the school; and should therefor make sure that the badge Url starts with the schoolUrl. Then the school url should be a constructor parameter.